api/tests/unit/features/test_unit_features_permissions.py

```diff
@@ -20,374 +21,453 @@
 
-        # When
-        result = feature_permissions.has_object_permission(
-            mock_request, mock_view, self.feature
-        )
+def test_project_user_without_view_project_permission_cannot_view_feature(
+    staff_user: FFAdminUser,
+    feature: Feature,
+) -> None:
+    # Given
+    feature_permissions = FeaturePermissions()
+    mock_view = mock.MagicMock(
+        kwargs={},
+        detail=True,
+        action="retrieve",
+    )
+    mock_request = mock.MagicMock(data={}, user=staff_user)
 
-        # Then
-        assert not result
+    # When
+    result = feature_permissions.has_object_permission(mock_request, mock_view, feature)
 
-    def test_organisation_admin_can_update_feature_segments(self):
-        # Given
-        mock_view.action = "segments"
-        mock_view.detail = True
-        mock_request.user = self.org_admin
+    # Then
+    assert result is False
 
-        # When
-        result = feature_permissions.has_object_permission(
-            mock_request, mock_view, self.feature
-        )
 
-        # Then
-        assert result
+def test_organisation_admin_can_edit_feature(
+    admin_user: FFAdminUser,
+    feature: Feature,
+) -> None:
+    # Given
+    feature_permissions = FeaturePermissions()
+    mock_view = mock.MagicMock(
+        kwargs={},
+        detail=True,
+        action="update",
+    )
+    mock_request = mock.MagicMock(data={}, user=admin_user)
 
-    def test_project_admin_can_update_feature_segments(self):
-        # Given
-        UserProjectPermission.objects.create(
-            user=self.user, project=self.project, admin=True
-        )
-        mock_view.action = "segments"
-        mock_view.detail = True
-        mock_request.user = self.user
+    # When
+    result = feature_permissions.has_object_permission(mock_request, mock_view, feature)
 
-        # When
-        result = feature_permissions.has_object_permission(
-            mock_request, mock_view, self.feature
-        )
+    # Then
+    assert result is True
 
-        # Then
-        assert result
 
-    def test_project_user_cannot_update_feature_segments(self):
-        # Given
-        mock_view.action = "segments"
-        mock_view.detail = True
-        mock_request.user = self.user
+def test_project_admin_can_edit_feature(
+    staff_user: FFAdminUser,
+    project: Project,
+    feature: Feature,
+) -> None:
+    # Given
+    UserProjectPermission.objects.create(user=staff_user, project=project, admin=True)
+    feature_permissions = FeaturePermissions()
+    mock_view = mock.MagicMock(detail=True, action="update")
+    mock_request = mock.MagicMock(user=staff_user)
 
-        # When
-        result = feature_permissions.has_object_permission(
-            mock_request, mock_view, self.feature
-        )
+    # When
+    result = feature_permissions.has_object_permission(mock_request, mock_view, feature)
+
+    # Then
+    assert result is True
+
+
+def test_project_user_cannot_edit_feature(
```

Comment:
This is a minor one but since the test name defines the user as a 'project user', perhaps they should have some affiliation with the project? Seems pointless, but just one to think about.

Reply:
That makes sense. I've added a `VIEW_PROJECT` permission to the user.